tests/Browser/LoginTest.php

```diff
@@ -0,0 +1,221 @@
+        });
+    }
+
+    /**
+     * Test login not input.
+     * Case 1-6
+     */
+    public function testLoginNotInput()
+    {
+        $this->browse(function ($browser) {
+            $browser->visit(config('contants.path.login'))
+                    ->press('Login')
+                    ->assertFocused('email');
+        });
+    }
+
+    /**
+     * Test email or password wrong input keep email.
+     * Case 1-7
+     */
+    public function testUserWrong()
+    {
+        // Get user
+        $user = User::find(2);
+        $this->browse(function ($browser) use ($user) {
+            $browser->visit(config('contants.path.login'))
+                    ->type('email', $user->email)
+                    ->type('password', 'abc@123')
+                    ->press('Login')
+                    ->assertSee('These credentials do not match our records.')
+                    ->assertInputValue('email', $user->email);
+        });
+    }
+
+    /**
+     * A basic browser test user login.
+     * Case 2-1
+     */
+    public function testUserLogin()
+    {
+        // Get user
+        $user = User::find(2);
+        $this->browse(function ($browser) use ($user) {
+            $browser->visit(config('contants.path.login'))
+                    ->type('email', $user->email)
+                    ->type('password', 'lifull@123')
+                    ->press('Login')
+                    ->assertPathIs('/');
+        });
+    }
+
+    /**
+     * A basic browser test user visit path admin.
+     * Case 2-2
+     */
+    public function testUserVisitAdmin()
+    {
+        $this->browse(function ($browser) {
+            $browser->loginAs(User::find(2))
+                    ->visit(config('contants.path.admin'))
+                    ->assertPathIs('/');
+        });
+    }
+
+    /**
+     * A basic browser test admin login.
+     * Case 2-3
+     */
+    public function testAdminLogin()
+    {
+        // Get Admin
+        $user = User::find(1);
+
+        $this->browse(function ($browser) use ($user) {
+            $browser->visit(config('contants.path.login'))
+                    ->type('email', $user->email)
+                    ->type('password', 'lifull@123')
+                    ->press('Login')
+                    ->assertPathIs(config('contants.path.admin'));
+        });
+    }
+
+    /**
+     * A basic browser test admmin visit path /.
+     * Case 2-4
+     */
+    public function testPathVisitUser()
+    {
+        $this->browse(function ($browser) {
+            $browser->loginAs(User::find(1))
+                    ->visit('/')
+                    ->assertPathIs(config('contants.path.admin'));
+        });
+    }
+
+    /**
+     * A basic browser test user logout.
+     * Case 3-1
+     */
+    public function testUserLogout()
+    {
+        $this->browse(function ($browser) {
+            $browser->loginAs(User::find(2))
+                    ->logout()
+                    ->assertGuest();
```

Comment:
test case check by login page , this assert is not

Reply:
i fixed in dev report

Reply:
expected and the test case are totally different , please fix

Reply:
i fixed